File: tests/fixtures/analyzers/javascript/callables.js
export function simple(value) {
    return value + 1;
}

export function longLinear(value) {
    const first = value + 1;
    const second = first + 1;
    const third = second + 1;
    const fourth = third + 1;
    const fifth = fourth + 1;
    const sixth = fifth + 1;
    return sixth;
}

export const arrow =
    (left, right) => {
        // Comment lines and blanks are physical source lines.

        return left + right;
    };

export const expression = function (value) {
    return value * 2;
};

export const helpers = {
    method(value) {
        return value;
    },
};

export class Worker {
    constructor(value) {
        this.value = value;
    }

    method(value) {
        return value;
    }
}

export function localOwner(value) {
    const local = (item) => item * 2;
    const localExpression = function (item) {
        return item + 1;
    };
    return local(localExpression(value));
}

export var
    legacy = function (value) {
        return value;
    };

File: tests/fixtures/analyzers/python/callables.py
def simple(value):
    return value + 1


def long_linear(value):
    first = value + 1
    second = first + 1
    third = second + 1
    fourth = third + 1
    fifth = fourth + 1
    sixth = fifth + 1
    return sixth


@audit
def documented(
    left,
    right,
):
    # Comment lines and blanks are physical source lines.

    return left + right


def outer(value):
    def local(item):
        return item * 2
    return local(value)


class Worker:
    def method(self, value):
        return value
